mods/default/functions.lua

```diff
@@ -327,31 +327,25 @@ end
 
 -- Leafdecay ABM
 
-minetest.register_abm({
-	label = "Leaf decay",
-	nodenames = {"group:leafdecay"},
-	neighbors = {"air"},
-	interval = 2,
-	chance = 10,
-	catch_up = false,
 
-	action = function(pos, node, _, _)
-		-- Check if leaf is placed
-		if node.param2 ~= 0 then
-			return
+function default.search_leaves_for_decay(pos, radius, lradius, trunk_node, leave_node)
+		local leaves_near = minetest.find_nodes_in_area(vector.subtract(pos, radius),
+				vector.add(pos, radius), leave_node)
+		for i = 1, #leaves_near do
+			local leaves_at_pos = leaves_near[i]
```

Comment:
We need to first check the param2 of the leaf, if it is non-zero (player placed) then 'continue' to the next leaf.
Worlds currently contain many player-placed leaves with param2 = 1 which should not decay or be removed.